From 318580922b7f6369299c2ae082850fa17a142d63 Mon Sep 17 00:00:00 2001
From: hscham <hscham@chromium.org>
Date: Thu, 3 Feb 2022 10:12:25 +0900
Subject: [PATCH] Add base::JSONReader options

Forward compatibility patch for r951523 uprev.
Note that only the new options are added and no actual change in
base::JSON{Reader,Parser} code to use them.

Change-Id: I1c90508350ab06fd2de5b81f0a6a3e2fc19cf632
---
 base/json/json_reader.h | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/base/json/json_reader.h b/base/json/json_reader.h
index b2821805fb..d72a5e3c00 100644
--- a/base/json/json_reader.h
+++ b/base/json/json_reader.h
@@ -62,6 +62,26 @@ enum JSONParserOptions {
   // not set, invalid code points trigger a hard error and parsing
   // fails.
   JSON_REPLACE_INVALID_CHARACTERS = 1 << 1,
+
+  // Allows both C (/* */) and C++ (//) style comments.
+  JSON_ALLOW_COMMENTS = 1 << 2,
+
+  // Permits unescaped ASCII control characters (such as unescaped \r and \n)
+  // in the range [0x00,0x1F].
+  JSON_ALLOW_CONTROL_CHARS = 1 << 3,
+
+  // Permits \\v vertical tab escapes.
+  JSON_ALLOW_VERT_TAB = 1 << 4,
+
+  // Permits \\xNN escapes as described above.
+  JSON_ALLOW_X_ESCAPES = 1 << 5,
+
+  // This parser historically accepted, without configuration flags,
+  // non-standard JSON extensions. This flag enables that traditional parsing
+  // behavior.
+  JSON_PARSE_CHROMIUM_EXTENSIONS = JSON_ALLOW_COMMENTS |
+                                   JSON_ALLOW_CONTROL_CHARS |
+                                   JSON_ALLOW_VERT_TAB | JSON_ALLOW_X_ESCAPES,
 };
 
 class BASE_EXPORT JSONReader {
-- 
2.35.0.rc2.247.g8bbb082509-goog

